File: apps/workers-api/src/card-image-url.ts
const CARD_IMAGE_ORIGIN = "https://image.tcgcard.fun";

export type CardImageVariant = "thumbnail" | "list" | "detail" | "preview" | "master";

export function cardImageUrl(cardRef: string, _variant: CardImageVariant): string {
  const objectPath = `cards/${encodeURIComponent(cardRef)}.jpg`;
  return `${CARD_IMAGE_ORIGIN}/${objectPath}`;
}

File: apps/flutter-app/lib/shared/card_image/card_image_url.dart
enum CardImageVariant { thumbnail, list, detail, preview, master }

const _cardImageOrigin = 'https://image.tcgcard.fun';

String cardImageUrl(String cardRef, CardImageVariant _) {
  final objectPath = 'cards/${Uri.encodeComponent(cardRef)}.jpg';
  return '$_cardImageOrigin/$objectPath';
}
